identity/aziot-dps-client-async/src/lib.rs

```diff
@@ -161,7 +158,7 @@ impl Client {
             .await;
         };
 
-        if matches!(auth_kind, DpsAuthKind::TpmWithAuth { .. }) {
+        if matches!(auth_kind, DpsAuthKind::TpmDpsNonce { .. }) {
```

Comment:
It seems we are recursing within `request()` itself. The nonce is imported into TPM when the `auth_kind` within `request()` method changes from `Tpm` to `TpmDpsNonce`. So, why are we reimporting the nonce into the TPM again after exiting `request()` back into `register()`?

Reply:
The first response from DPS contains a challenge key. The second response from DPS contains the symmetric key registered with IoT Hub.

Reply:
Ah yes. Thanks.